packages/react-dialog/src/components/Dialog/renderDialog.tsx

```diff
@@ -1,13 +1,19 @@
 import * as React from 'react';
 import { getSlots } from '@fluentui/react-utilities';
 import type { DialogState, DialogSlots } from './Dialog.types';
+import { Portal } from '@fluentui/react-portal';
 
 /**
  * Render the final JSX of Dialog
  */
 export const renderDialog_unstable = (state: DialogState) => {
   const { slots, slotProps } = getSlots<DialogSlots>(state);
+  const defaultOverlay = state.type !== 'non-modal' && <div aria-hidden="true" className={state.overlayClassName} />;
 
-  // TODO Add additional slots in the appropriate place
-  return <slots.root {...slotProps.root} />;
+  return state.open ? (
+    <Portal>
+      {slots.overlay ? <slots.overlay {...slotProps.overlay} /> : defaultOverlay}
+      <slots.root {...slotProps.root}>{slotProps.root.children} </slots.root>
+    </Portal>
+  ) : null;
```

Comment:
```suggestion
  if (!state.open) {
    return null;
  }
  return (
    <Portal>
      {slots.overlay ? <slots.overlay {...slotProps.overlay} /> : defaultOverlay}
      <slots.root {...slotProps.root}>{slotProps.root.children} </slots.root>
    </Portal>
  ) ;
```

nit: I think this is cleaner